accesspoc/dips/models.py

```diff
@@ -1,6 +1,10 @@
+from abc import ABCMeta, abstractmethod
```

Comment:
Playing dumb a bit but there is an interesting mix of Python OOP paradigms and Pythonisms in this module that I don't fully understand. Would it be possible to add a docstring at the module level explaining your solution with  and what are the consequences.

Reply:
Will do!

Mainly, I don't want a new ES related model being created and not failing until ES access is attempted because it lacks some property/method, that's too much duck typing for me these days.